java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -5769,6 +5769,68 @@ public void bzmpop_binary_timeout_check(BaseClient client) {
         }
     }
 
+    @SneakyThrows
+    @ParameterizedTest(autoCloseArguments = false)
+    @MethodSource("getClients")
+    public void xadd_duplicate_entry_keys(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+        String field = UUID.randomUUID().toString();
+        String foo1 = "foo1";
+        String bar1 = "bar1";
+
+        String timestamp = "0-1";
+        assertEquals(
+                timestamp,
+                client
+                        .xadd(
+                                key,
+                                new String[][] {new String[] {field, foo1}, new String[] {field, bar1}},
+                                StreamAddOptions.builder().id(timestamp).build())
+                        .get());
+
+        // get everything from the stream
+        Map<String, String[][]> result = client.xrange(key, InfRangeBound.MIN, InfRangeBound.MAX).get();
+        assertEquals(2, result.size());
+        String[][] entry = result.get(key);
```

Comment:
I think you can create `entry` which pass to `xadd` and the compare it with received one.
Use `assertDeepEquals` to compare the entire content at once.